questionnaire_linking.py

```diff
@@ -0,0 +1,136 @@
+            return qid_file(qid_link_file)
+    except UnicodeDecodeError as err:
+        print(f'Invalid file encoding, requires utf-8, error: {err}')
+        exit(-1)
+
+
+def qid_file(qid_link_file):
+    qid_link_file_reader = csv.DictReader(qid_link_file, delimiter=',', fieldnames=['case_ref', 'qid'])
+    validate_and_submit_questionnaire_links(qid_link_file_reader)
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description='Load a QID Link file into response management')
+    parser.add_argument('qid_link_file_path', help='path to the QID Link file', type=str)
+    return parser.parse_args()
+
+
+def validate_and_submit_questionnaire_links(qid_link_file_reader):
+    global CASE_REF_ERROR_COUNT
+    global QID_ERROR_COUNT
+    for line_number, questionnaire_link in enumerate(qid_link_file_reader, 1):
+        CASE_REF_ERROR_COUNT = 0
+        QID_ERROR_COUNT = 0
+        for i in questionnaire_link:
+            questionnaire_link[i] = questionnaire_link[i].replace(" ", "")
+        validate_qid_len_and_type(line_number, questionnaire_link["qid"])
+        validate_qid_third_digit(line_number, questionnaire_link["qid"])
+        validate_check_digits(line_number, questionnaire_link["qid"])
+        case_id = validate_case_ref(line_number, questionnaire_link["case_ref"])
+
+        if CASE_REF_ERROR_COUNT or QID_ERROR_COUNT:
+            print(f'Error: Line {line_number}: {questionnaire_link} has FAILED')
+        else:
+            post_message_to_queue(case_id, questionnaire_link["qid"], line_number)
+        print("\n")
+
+
+def validate_case_ref(line_number, case_ref):
+    global CASE_REF_ERROR_COUNT
+    if len(case_ref) != 8:
+        print(f'Error: Line {line_number}: Incorrect length of Case Reference {case_ref}. Expected length is 8 digits')
+        CASE_REF_ERROR_COUNT += 1
+    if not case_ref.isdigit():
+        print(f'Error: Line {line_number}: Incorrect type of Case Reference {case_ref}. Case Reference MUST be numeric')
+        CASE_REF_ERROR_COUNT += 1
+    if CASE_REF_ERROR_COUNT == 0:
+        response = requests.get(f'http://{Config.CASEAPI_HOST}:{Config.CASEAPI_PORT}/cases/ref/{case_ref}')
+        if response.status_code == 404:
+            print(f'Error: Line {line_number}: Case Reference {case_ref} does not exist')
+            CASE_REF_ERROR_COUNT += 1
+            return None
+        response.raise_for_status()
+        return response.json()["id"]
+
+
+def validate_qid_len_and_type(line_number, qid):
+    global QID_ERROR_COUNT
+    if len(qid) != 16:
+        print(f'Error: Line {line_number}: Incorrect length of QID {qid}. Expected length is 16 digits')
+        QID_ERROR_COUNT += 1
+    if not qid.isdigit():
+        print(f'Error: Line {line_number}: Incorrect type of QID {qid}. QID MUST be numeric')
+        QID_ERROR_COUNT += 1
+
+
+def validate_qid_third_digit(line_number, qid):
+    global QID_ERROR_COUNT
+    if len(qid) >= 3:
+        if qid[2] != "2":
```

Comment:
Could just be 
```python
if len(qid) >= 3 and qid[2] != "2":
```
`and` is short circuited in python.

Reply:
Sorted